protocol/cosi.go

```diff
@@ -74,6 +75,9 @@ type CommitmentHook func(in []abstract.Point) error
 // challenge is received
 type ChallengeHook func(ch abstract.Scalar) error
 
+// ResponseHook allows registering a handler when the response is calculated
```

Comment:
The response hook is not receiving the calculated response but the children's response.